core/src/main/java/com/linecorp/armeria/server/healthcheck/CpuHealthChecker.java

```diff
@@ -0,0 +1,180 @@
+final class CpuHealthChecker implements HealthChecker {
+    private static final Logger logger = LoggerFactory.getLogger(CpuHealthChecker.class);
+
+    private static final MethodHandles.Lookup LOOKUP = MethodHandles.lookup();
+
+    private static final DoubleSupplier currentSystemCpuUsageSupplier;
+
+    private static final DoubleSupplier currentProcessCpuUsageSupplier;
+
+    @Nullable
+    private static final OperatingSystemMXBean operatingSystemBean;
+
+    private static final Class<?> operatingSystemBeanClass;
+
+    private static final List<String> OPERATING_SYSTEM_BEAN_CLASS_NAMES = ImmutableList.of(
+            "com.ibm.lang.management.OperatingSystemMXBean", // J9
+            "com.sun.management.OperatingSystemMXBean" // HotSpot
+    );
+
+    @Nullable
+    private static final MethodHandle systemCpuLoad;
+
+    @Nullable
+    private static final MethodHandle processCpuLoad;
+
+    static {
+        operatingSystemBeanClass = requireNonNull(getFirstClassFound(OPERATING_SYSTEM_BEAN_CLASS_NAMES));
+        operatingSystemBean = ManagementFactory.getOperatingSystemMXBean();
+        final MethodHandle getCpuLoad = detectMethod("getCpuLoad");
+        systemCpuLoad = getCpuLoad != null ? getCpuLoad : detectMethod("getSystemCpuLoad");
+        processCpuLoad = detectMethod("getProcessCpuLoad");
+        currentSystemCpuUsageSupplier = () -> invoke(systemCpuLoad);
+        currentProcessCpuUsageSupplier = () -> invoke(processCpuLoad);
+    }
+
+    private final DoubleSupplier systemCpuUsageSupplier;
+
+    private final DoubleSupplier processCpuUsageSupplier;
+
+    @VisibleForTesting
+    final double targetProcessCpuLoad;
+
+    @VisibleForTesting
+    final double targetSystemCpuUsage;
+
+    /**
+     * Instantiates a new Default cpu health checker.
+     *
+     * @param cpuUsageThreshold the target cpu usage
+     * @param processCpuLoadThreshold the target process cpu usage
+     */
+    private CpuHealthChecker(double cpuUsageThreshold, double processCpuLoadThreshold) {
+        this(cpuUsageThreshold, processCpuLoadThreshold,
+                currentSystemCpuUsageSupplier, currentProcessCpuUsageSupplier);
+    }
+
+    private CpuHealthChecker(double targetSystemCpuUsage, double targetProcessCpuLoad,
+                             DoubleSupplier systemCpuUsageSupplier, DoubleSupplier processCpuUsageSupplier) {
+        checkArgument(targetSystemCpuUsage >= 0 && targetSystemCpuUsage <= 1.0,
+                      "cpuUsage: %s (expected: 0 <= cpuUsage <= 1)", targetSystemCpuUsage);
+        checkArgument(targetProcessCpuLoad >= 0 && targetProcessCpuLoad <= 1.0,
+                      "processCpuLoad: %s (expected: 0 <= processCpuLoad <= 1)", targetProcessCpuLoad);
+        this.targetSystemCpuUsage = targetSystemCpuUsage;
+        this.targetProcessCpuLoad = targetProcessCpuLoad;
```

Comment:
so that others aren't confused about the difference of "load" and "usage"

```suggestion
        this.targetProcessCpuUsage = targetProcessCpuUsage;
```